### DISCUSSION TITLE
Adapting the Mujoco model to enable drifting

### DISCUSSION THREAD

**Discussion by silverjoda** [prl-mushr/mushr#59] [General]
Dear Mushr team,
 
I'm currently working with a student on a sim-to-real project for a buggy car and recently came across the 
Mujoco model that you have in your project. I had issues trying to make it drift (wheel slip) so I opened a [issue](https://github.com/deepmind/mujoco/issues/57) on the mujoco
github page, where the authors were kind enough to help me solve my issue so I am here to inform of a subtle problem that exists with the model, and also how to set parameters of the model so that one can do drifts with it.

The issue: 
If you lower the wheel friction then the car starts to exhibit strange behavior. This is caused by the following:
1) The mesh used for the wheels is not well centered and causes wheel bouncing
2) There is no suspension which leads to uneven forces on the wheels
3) The contacts are very hard which causes bouncing.  

The solution:
1) Use a correctly scaled and centered ellipsoid as the collision instead of a mesh
2) I added a very simple spring damper to each wheel
3) setting the first parameter on solimp to zero makes the contacts soft. This has to be present on both contacting bodies to work apparently.

If anyone is interested, here is the modified models directory that you can just copy paste into the package. I suggest using the mujoco "simulate" binary which you can just run and then click and drop the one_car.xml model into the simulation, press "c" to see contacts and use the sliding bar for actuation.
[models.zip](https://github.com/prl-mushr/mushr/files/7594314/models.zip)

https://user-images.githubusercontent.com/21022992/143213219-222548b3-ee7b-4302-9f83-cdcc4183d736.mp4

Thanks, T

**Comment by palanc** [comment]
Thanks so much for letting us know @silverjoda - I've created an issue for it and we will work on integrating your solution.
